loop/mi_views.py

```diff
@@ -0,0 +1,189 @@
+
+        # Store Json Result
+        agg_res = []
+
+        # Mandi Detail
+        mandi_res = []    
+        for mandiobj in mandi_data_obj:
+            mandiobjdetail = MandiDetail(mandi_id=mandiobj['id'], mandi_name=mandiobj[mandi_name_label], mandi_category='Chhoti Mandi',\
+                        mandi_distance='590')
+            gaddidar_obj= gaddidar_data_obj.filter(mandi__id=mandiobj['id']).values('id', gaddidar_name_label, 'gaddidar_phone')
+            for gaddidarobj in gaddidar_obj:
+                gaddidarobj = GaddidarDetail(gaddidar_id=gaddidarobj['id'], gaddidar_name=gaddidarobj[gaddidar_name_label],\
+                            gaddidar_phone_no=gaddidarobj['gaddidar_phone'])
+                mandiobjdetail.gaddidar_list.append(gaddidarobj.__dict__)
+            
+            transport_dataframe_obj = transport_dataframe[transport_dataframe['Mandi Id'] == mandiobj['id']]
+            for index, row in transport_dataframe_obj.iterrows():
+                transportobj = TransportDetail(transport_id=row['Vehicle Id'], transport_name=row['Vehicle Name'], transport_cost=str(row['Cost']),\
+                                transport_capacity=row['Capacity'])
+
+                mandiobjdetail.distance = row['Mandi Distance'] 
+                mandiobjdetail.category = row['Mandi Category']
+                mandiobjdetail.transport_list.append(transportobj.__dict__)
+
+            mandi_res.append(mandiobjdetail.__dict__)
+
+        # Aggregator Details
+        for aggobj in agg_data_obj:
+            aggobj = AggregatorDetail(aggregator_id=aggobj['user'], aggregator_name=aggobj['name_en'])
+            aggobj.mandi_list = mandi_res
+            agg_res.append(aggobj.__dict__)
+
+        data = json.dumps(agg_res)
+        return HttpResponse(data)
+    else :
+        return HttpResponse(status=401)
+    
+
+def get_crop_prices(request):
+
+    agg_list_requested = is_authenticated(request)
+    
+    if agg_list_requested :
+        # Handling Request Params
+        kwargs = read_params(request)
+
+        cropobj = Crop.objects.values_list('id', flat=True)
+        cropobj = map(int, cropobj)
+        # Prepare data
+        crop_list = tuple(cropobj)
+
+        query = get_query.query_for_rates(crop_list , mandi_list, date_range=kwargs['day_limit'])
+        
+        result = run_query(query)
+        dataframe = remove_crop_outliers(ct_data = result)
+
+        # Grouping Raw Data to Crop & Mandi wise.
+        df = dataframe.groupby(['Crop', 'Market_Real'])
+
+        # Creation of JSON
+        res = []
+        for i, r in df:
+            # Assigning min date
+            latest_price_date = pd.Timestamp('2015-01-01')
+            cropmandidata = CropMandiData(crop_id=i[0], mandi_id= i[1])
+            for index, row in r.iterrows():
+                crop, mandi, date, Av_Rate, STD, PriceMax, PriceMin = row['Crop'], row['Market_Real'], row['Date'], row['Av_Ratemean'], row['STDmean'], row['Pricemax'], row['Pricemin']
```

Comment:
Are we sure that avg_price and std are always returned from the function?

Reply:
You doubt that? :<, BTW yes.